### PR TITLE
add ability to pass in a scope instead of hardcoding

### PR DESCRIPTION
**Pull request by chadhutchins**
First off... thanks for this package! We use it to connect on behalf of multiple Zendesk users, which requires us to set different scopes when we instantiate the provider. This PR allows us to set the scope through a setDefaultScopes() method. Please consider, thanks!

### CODE REVIEW

**Comment by chadhutchins** [comment]
It doesn't appear the CI build issues are related to any of my code, but I'm happy to investigate further if I need to.

**Comment by chadhutchins** [comment]
Hi @stevenmaguire can you please consider this PR. Thanks!

**Comment by stevenmaguire** [comment]
@chadhutchins thanks for the PR. Scopes are only required during the composition of the authorization URL phase and are not a provider-specific configuration. The default scopes defined in this package are only used if you do not provide any scopes during the authorization URL composition. 

This package already supports adding scopes when composing the authorization URL. You can see this functionality exercised in the test suite: 

https://github.com/stevenmaguire/oauth2-zendesk/blob/8e9b50620712f23fcb824606a4c89c475e9e24f3/tests/src/Provider/ZendeskTest.php#L82-L90

I realize now that the README does not include an example of this functionality, so I'm happy to accept a PR that updates the README documentation to include this.

File: src/Provider/Zendesk.php [hunk 1]
```diff
@@ -81,7 +83,24 @@ public function getResourceOwnerDetailsUrl(AccessToken $token)
...
+
+    /**
+     * Set the default scopes used by this provider.
+     *
+     * This should not be a complete list of all scopes, but the minimum
+     * required for the provider user interface!
+     *
+     * @return Zendesk
+     */
+    public function setDefaultScopes(Array $scopes)
```
Comment (stevenmaguire) [inline src/Provider/Zendesk.php:97]: @chadhutchins As an FYI, this line was responsible for the CI failure...

```
----------------------------------------------------------------------
549FOUND 1 ERROR AFFECTING 1 LINE
550----------------------------------------------------------------------
551 97 | ERROR | [x] PHP keywords must be lowercase; expected "array"
552    |       |     but found "Array"
553----------------------------------------------------------------------
554PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
555----------------------------------------------------------------------
```

There is a code style linter that runs as a part of the build.

**Comment by chadhutchins** [comment]
Well... you're spot on! I banged my head on this forever. Passing in the scope array to the getAuthorizationUrl method definitely does the trick. Thanks for reviewing this and thanks for the CI failure tip.

**Comment by stevenmaguire** [comment]
You're welcome! I appreciate you contributing.

**Pull request closed without merging**